### ISSUE TITLE
fromJson (toJson .Chart) does not have reflexive behavior

### ISSUE THREAD

**Issue by SOF3** [helm/helm#12990]
Output of `helm version`:

```
version.BuildInfo{Version:"v3.14.2", GitCommit:"c309b6f0ff63856811846ce18f3bdc93d2b4d54b", GitTreeState:"clean", GoVersion:"go1.21.7"}
```

When I evaluate `.Chart.Name`, it goes through the Go exported field. But when I evaluate `(fromJson (toJson .Chart)).Name`, it becomes null, because `toJson` emits `Name` as the alias `name` instead, and `fromJson` would not unmarshal it back to the tagged struct and I have to write `.name` instead of `.Name`.

**Comment by SOF3** [comment]
I know it was weird of me to `toJson` a `.Chart`. I did so simply because I was passing an enormous dictionary as argument to another template, and I thought it was easier to format it as YAML and `(include "internal-template" . | include "library-template")`, and I tried to write the `.Chart` inside the YAML.

**Comment by banjoh** [comment]
@SOF3 do you have a complete, but minimal example which I can look at and test the failure scenario? It's a bit unclear (to me) what is happening here.

**Comment by SOF3** [comment]
```console
$ cat templates/example.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: abc
data:
  release: {{include "show-keys" (toJson .Release) | toJson}}
  chart: {{include "show-keys" (toJson .Chart) | toJson}}

{{- define "show-keys" -}}
{{keys (fromJson .)}}
{{- end}}

$ helm template .
---
# Source: demo/templates/example.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: abc
data:
  release: "[Revision Service IsInstall IsUpgrade Name Namespace]"
  chart: "[version description apiVersion appVersion type IsRoot name]"
```

**Comment by SOF3** [comment]
This issue is still valid, and it is not a support request.

**Comment by JuryA** [comment]
`.Chart` and `.Release` are `structs` ... this will not work as expected. You need to convert it first to `map[string]{}`.

**Comment by SOF3** [comment]
yeah and this is a bit counterintuitive. Would it be a good idea to make the marshal/unmarshal of these structs use AsMap by default? It doesn't make sense to couple users' helm chart logic (especially generic code that doesn't know whether it is a struct or a map) with helm's internal implementation details (structs vs maps).

**Comment by SOF3** [comment]
Is the disposition to close this issue due to any possible fix going to break BC, or is it planned to change this?
